src/basic/hostname-util.c

```diff
@@ -11,6 +12,21 @@
 #include "string-util.h"
 #include "strv.h"
 
+const char* get_fallback_hostname(void) {
+        const char *e;
+
+        e = secure_getenv("SYSTEMD_FALLBACK_HOSTNAME");
+        if (e) {
+                if (!hostname_is_valid(e, 0))
+                        log_debug("Invalid hostname in $SYSTEMD_FALLBACK_HOSTNAME, ignoring.");
+                else
+                        return e; /* The pointer is valid as long as a getenv(), putenv(), setenv(), or
```

Comment:
Sounds a bit dangerous, given it's an (internal) API rather than just a static helper. I totally see myself falling flat on this :-) Is there any perf critical path where this would be beneficial, instead of just strdup'ing it?

Reply:
I think it's fine. All callers immedatiely strdup() the variable. But anyway, it's reworked in one of the later patches.

Reply:
yep saw that now, was going by commit (and then got interrupted by meetings ofc)